sources/quest_3/chapter_10/step_28/step_28_start.move

```diff
@@ -0,0 +1,182 @@
+            nb_tree: 10,
+            has_river: true,
+            shore_location: @0x42,
+            daily_visitors: vec,
+            island_name: utf8(b"SHUJU"), 
+            nb_house: 1,   
+        }; 
+    }
+    
+    fun resource_day() : (u64, u64){
+        let food_day: u64 = 10;
+        let log_day: u64 = 5;
+        (food_day, log_day)
+    }
+
+    fun check_resource(): bool {
+        let (daily_food, daily_log) = resource_day();
+        if (daily_food == 10 && daily_log != 6 ){
+            true
+        }
+        else{
+            false
+        }
+    }
+
+    fun check_resourceShortage(r: &Resources){
+        let (daily_food, daily_log) = resource_day();
+        let (total_food, total_log) = resources_avail(r);
+        assert!((total_food>= daily_food && total_log>=daily_log), error:: not_found(E_RESOURCE_SHORTAGE));
+    }
+    
+    // fun add_member(h: &mut House){ 
+    //    if(h.no_of_members >= 4)
+    //   {
+    //       abort 0;
+    //   }
+    //   h.no_of_members = h.no_of_members + 1;
+    // }
+
+    fun add_member(h: &mut House){ 
+        assert! (h.no_of_members >= 4, 0);
+        h.no_of_members = h.no_of_members + 1;
+    }
+    
+   fun change_name(new_name: String, data: &mut GlobalData, s: signer){
+        assert!(signer::address_of(&s) == @0x42, 0);
+        let byte: vector<u8> = bcs::to_bytes(&new_name);
+        data.island_name = utf8(byte);
+    }
+
+    fun build_house(data: &mut GlobalData){
+        if(data.nb_tree < 5){
+            data.nb_tree = data.nb_tree + 1;
+        }
+        else {
+            data.nb_house = data.nb_house + 1;
+            data.nb_tree = data.nb_tree - Min_trees;
+        }
+    }
+
+     fun print_welcomeMessage(){
+        let welcomeMessage = utf8(b"Welcome to our Island");
+        print(&welcomeMessage);
+    }
+
+    fun print_dailyVisitors_usingloop(data: &GlobalData){
+        let vec:vector<u64> = data.daily_visitors;
+        let len:u64 = vector::length(&vec);
+        let i = 0;
+        loop {
+            if (i < len) continue;
+            if (i >= len) break;
+            let visitor_today = vector::borrow(&vec, i);
+            print(visitor_today);
+            i = i + 1;
+        };
+    }
+
+    fun print_dailyVisitors_usingwhile(data: &GlobalData){
+        let vec:vector<u64> = data.daily_visitors;
+        let len:u64 = vector::length(&vec);
+        let i = 0;
+        while (i < len) {
+            let visitor_today = vector::borrow(&vec, i);
+            print(visitor_today);
+        };
+    }
+
+    fun createVisitorID(info: &mut VisitorInfo<u64, u64>, visitor_age: u64) {
+       let currentID = info.id;
+       let nextID: u64 = currentID + 1;
+       let _info = VisitorInfo<u64, u64> {
+            id: nextID,
+            age: visitor_age,
+        };
+    }
+
+    //create a public(friend) function `daily_visitors()` of type u64 that returns the value `6`
+    //create a module named `tour_guide` under the package `robinson`
```

Comment:
This should be after the `}`, instructions should goes to place user have to use

Reply:
done